storage/src/main/scala/ch/epfl/bluebrain/nexus/storage/attributes/ContentTypeDetector.scala

```diff
@@ -0,0 +1,45 @@
+package ch.epfl.bluebrain.nexus.storage.attributes
+
+import akka.http.scaladsl.model.{ContentType, MediaType, MediaTypes}
+import akka.http.scaladsl.model.HttpCharsets.`UTF-8`
+import akka.http.scaladsl.model.MediaTypes.{`application/octet-stream`, `application/x-tar`}
+import ch.epfl.bluebrain.nexus.delta.kernel.http.MediaTypeDetectorConfig
+import ch.epfl.bluebrain.nexus.delta.kernel.utils.FileUtils
+
+import java.nio.file.{Files, Path}
+import scala.util.Try
+
+final class ContentTypeDetector(config: MediaTypeDetectorConfig) {
+
+  /**
+    * Detects the media type of the provided path, based on the custom detector, the file system detector available for
+    * a certain path or on the path extension. If the path is a directory, a application/x-tar content-type is returned
+    *
+    * @param path
+    *   the path
+    * @param isDir
+    *   flag to decide whether or not the path is a directory
+    */
+  def apply(path: Path, isDir: Boolean = false): ContentType =
```

Comment:
I guess it could be wrapped in a `F[_]`/ an `IO` but it feels like enough yak shaving and enough complexity for a single PR